machineconfiguration/v1alpha1/doc.go

```diff
@@ -0,0 +1,8 @@
+// +k8s:deepcopy-gen=package,register
+// +groupName=machineconfiguration.openshift.io
+// +k8s:defaulter-gen=TypeMeta
+// +k8s:openapi-gen=true
+
+// +kubebuilder:validation:Optional
```

Comment:
Please don't set package level optional, it's more likely to lead to confusion than it is to help. Each field should explicitly set optional or required anyway